src/App.vue

```diff
@@ -351,6 +362,34 @@ const { isOverDropZone } = useDropZone(dropZoneRef, {
   // whether to prevent default behavior for unhandled events
   preventDefaultForUnhandled: false,
 })
+
+// save spec to localStorage
+watchDebounced(specText, (newValue) => {
+  if (!isCleared.value && newValue && newValue.trim() !== defaultSpec.trim()) {
+    saveSpecToLocalStorage({
+      key: STORAGE_KEY,
+      value: newValue,
+      onFallback: (error) => {
+        toaster.open({
+          appearance: 'danger',
+          message: 'Failed to save specification to localStorage.',
+        })
```

Comment:
I don't think we need a toast notification here; the user likely doesn't know or care about the localStorage implementation